src/payment/payment.service.ts

```diff
@@ -1,26 +1,71 @@
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable, Req } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { User } from 'src/user/entities/user.entity';
+import { UserInterface } from 'src/user/interfaces/user.interface';
 import { CreatePaymentDto } from './dto/create-payment.dto';
 import { UpdatePaymentDto } from './dto/update-payment.dto';
+import { PaymentInterface } from './interfaces/payment.interface';
 
 @Injectable()
 export class PaymentService {
-  create(createPaymentDto: CreatePaymentDto) {
-    return 'This action adds a new payment';
+  constructor(
+    @InjectModel('Payment') private readonly paymentModel: Model<PaymentInterface>,
+    @InjectModel('User') private readonly userModel: Model<UserInterface>,
+  ) {}
+
+  async create(createPaymentDto: CreatePaymentDto) {
+    const user = await this.userModel.findById(createPaymentDto.user).exec();
+
+    if (user) {
+      const payment = await new this.paymentModel(createPaymentDto);
+      return payment.save();
+    } else {
+      throw new HttpException('User not found', HttpStatus.NOT_FOUND);
+    }
+  }
+
+  async findAll() {
+    const payment = await this.paymentModel.find().exec();
+    if (!payment || !payment[0]) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return payment;
   }
 
-  findAll() {
-    return `This action returns all payment`;
+  async findAllPaymentsForUser(userId: User) {
+    const payment = await this.paymentModel.find({ User }).exec();
```

Comment:
Jak czytam dokumentacje mongoose to tutaj wydaje się że źle urzywasz tej funkcji find , sprawdzałeś sobie to w postmanie ??
https://mongoosejs.com/docs/api.html#model_Model.find

Reply:
Tak, wszystkie endpointy działają.

Reply:
oki

Reply:
- tutaj nazewnictwo jest trochę mylące, bo jako parametr jest `userId` a typ `User`
- teraz jeśli się wyszukuje jakąś płatność to w userze jest przechowywane tylko id usera, a chcemy tam mieć usera tylko bez hasła i płci, bo w płatności nam to nie jest potrzebne
- jak się doda `populate('user', '-password -gender')` a trzeba to dodać we wszystkich findach to będzie błąd w tej funkcji, fajnie by było jakbyś zerknął jak to jest w role/ pass, żeby też był ten kod w miarę spójny 
- teraz ten `User` w find jest trochę bez sensu, bo nawiasy `{}` w `()` oznaczają destrukturyzację obiektu, a tu przekazujesz cały obiekt (a chyba nawet nie obiekt tylko zaimportowany schemat?)